cgsmiles/resolve.py

```diff
@@ -69,62 +68,80 @@ def match_bonding_descriptors(source, target, bond_attribute="bonding"):
 
 class MoleculeResolver:
     """
-    Resolve the molecule described by a CGBigSmile
-    string.
+    Resolve the molecule(s) described by a
+    CGSmiles string. Each execution of the
+    resolve function will return the next
+    resolved molecule and the graph of the
+    previous resolution.
+
+    Use the resolve_iter method to loop over
+    all possible molecule resolutions enconded
+    in the CGSmiles string.
     """
-
     def __init__(self,
                  pattern,
                  meta_graph=None,
                  fragment_dict={},
-                 all_atom=True):
+                 last_all_atom=True):
 
-        # let's start by setting some attributes
-        # this is the fragment string
-        self.fragment_str = None
-        # this is the cgsmiles string
-        self.cgsmiles_str = None
+        """
+        Parameters
+        ----------
+        pattern: str
+            the cgsmiles string to resolve
+        meta_graph: `:class:nx.Graph`
+            a potential higher resolution graph
+        fragment_dict: dict[`:class:nx.Garph`]
+            a dict of fragment graphs
+        last_all_atom: bool
+            if the last resolution is at the all
+            atom level. Default: True
+        """
         # this is the dict with the fragments
         # either provided and/or extracted
         # from the fragment string
         self.fragment_dict = fragment_dict
-        # this is the lower resolution graph
-        self.meta_graph = meta_graph
-        # this is the higher resolution graph
+        # this is the current resolution graph
         self.molecule = nx.Graph()
-        # this attribute stores if the fragments
-        # follow open_smiles syntax or cg_smiles
-        # syntax
-        self.all_atom = all_atom
+        # this is the current meta_graph
+        self.meta_graph = nx.Graph()
+        # these are possible fragment strings
+        self.fragment_strs = []
 
-        # here we figure out what we are dealing with
+        # here we figure out how many resolutions
+        # we are dealing with
         elements = re.findall(r"\{[^\}]+\}", pattern)
+
```

Comment:
Is `{}` allowed? If so, change the `+` to a `*`

Reply:
Nope that's not allowed.